### ISSUE TITLE
Vim 8.1 (1-899): Illegal Variable Name a:attr, a:guisp

### ISSUE THREAD

**Issue by jmzagorski** [chriskempson/base16-vim#197]
Hi and thanks for this great plugin in addition to the terminal colors:

I have not compiled vim from source in a while (so this could be my fault if I have overlooked a missing feature), but I did last night on my windows 10 machine. Everything works fine after the compilation except this plugin throws errors when opening vim. The messages are as follows:

![image](https://user-images.githubusercontent.com/8404957/52716274-6b9dfc00-2f6c-11e9-80c8-79685738dd91.png)

![image](https://user-images.githubusercontent.com/8404957/52716689-52497f80-2f6d-11e9-90b6-15a87216b162.png)

I am using the base16-eighties colorscheme and this is my vimrc below:

**vimrc**
```
call dein#add('chriskempson/base16-vim')

if filereadable(expand("~/.vimrc_background"))
  "per the docs only enable this if the ~/.cofig/base16/colortest does not show blue for 17-21
  let base16colorspace=256
  source ~/.vimrc_background
endif
```

**vimrc_background**
```
if !exists('g:colors_name') || g:colors_name != 'base16-eighties'
  colorscheme base16-eighties
endif
```
When i remove this from my vimrc vim opens and works fine

**Comment by steeef** [comment]
I'm seeing this too. Looks like this is the commit that broke this plugin:
https://github.com/vim/vim/commit/31b816042fca879b11965ddd75287732563ba698

so we're not supposed to be using the `a:` dict as it should be immutable. Would renaming it everywhere fix this?

**Comment by steeef** [comment]
Here's the issue that brought this up: https://github.com/vim/vim/pull/3929

**Comment by steeef** [comment]
Attempted just changing references for `let a:guisp` and `let a:attr` within the `function! g:Base16hi`, but it broke the colorscheme from what I can see. There's something about that dict and its immutability that's throwing me off.

**Comment by jmzagorski** [comment]
Thanks for pointing me to those links. If I have time I will try and mess around with the files too. For now I will build prior to that patch version.

**Comment by chrisjeon** [comment]
Following. Is there a temporary patch for this?

**Comment by jmzagorski** [comment]
For now i would not build vim past patch 887. I rebuilt on 887 and everything is fine.

**Comment by chrisjeon** [comment]
How would you go back a version on Mac with homebrew?

**Comment by jsok** [comment]
@chrisjeon I had the same issue, the easiest way is to install the formula url like:

‘’’
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/$REVISION/Formula/vim.rb
‘’’

Browse the GitHub repo to find the revision matching the vim version you wish to downgrade to.

**Comment by chrisjeon** [comment]
To install the previous version, what does the $REVISION part have to be?

**Comment by peikk0** [comment]
`a74354c28ae78659451789179c17b8240d837b94` is good (https://github.com/Homebrew/homebrew-core/commit/a74354c28ae78659451789179c17b8240d837b94)

**Comment by sebmaynard** [comment]
@chriskempson please could you consider merging in the change @danielwe suggested, it's a bit of a shame to see more people's vimrc switching over to an unofficial fork

**Comment by kmARC** [comment]
If you don't want to use others' fork, as a quick-and-dirty replacement on your local copy can be done by some vim fun 😊

```
cd ~/.vim/plugged/base16-vim/colors # should you use vim-plug

vim *.vim # set up the arglist
```

Then in vim:
```
:argdo g!/Base16hi/s/a:\(attr\|guisp\)/l:\1/g
:wa
```

This replaces all `a:attr`, and `a:guisp` to their `l:` counterpart on al the lines except for the `Base16hi` invocation, in all the files.

**Comment by sebmaynard** [comment]
Nice idea @kmARC 

If you want this to happen automagically when vim-plug installs the base16-vim plugin...

    function FixupBase16(info)
        !sed -i '/Base16hi/\! s/a:\(attr\|guisp\)/l:\1/g' ~/.vim/plugged/base16-vim/colors/*.vim
    endfunction
    Plug 'chriskempson/base16-vim', { 'do': function('FixupBase16') }

Now I can continue using the official repo with a minor tweak I have control over, but never have to think about it.

**Comment by raskchanky** [comment]
@sebmaynard You're a gentleman and a scholar.  Thank you for that.

**Comment by kmARC** [comment]
> Now I can continue using the official repo with a minor tweak I have control over, but never have to think about it.

@sebmaynard Are you sure vim-plug's `git pull` would work automatically? Your working copy will have now unstaged changes.

**Comment by szero** [comment]
My temporary fix for this problem:
```sh
cd <your_path_to_base16-vim>
git remote add dict_fix https://github.com/danielwe/base16-vim
git pull dict_fix master:dictfix
git checkout dictfix
```
When the PR will get merged, one can just remove the remote, checkout the `master` branch and delete the `dictfix` one.

**Comment by iamsubhranil** [comment]
@szero i like yours better.

**Comment by padde** [comment]
If you want to use the workaround by @sebmaynard on macOS, you need to `brew install gnu-sed` and replace `sed` with `gsed`. The BSD `sed` shipped with macOS will not work with the given command.

**Comment by wincent** [comment]
Another possible workaround: the whole assignment-to-parameters thing was only fairly recently introduced, in 9ba1ad4b2e8a7190d2c5c3940252c4a1282485f0 (there are only two other non-merge commits after this one on "master"). So if you want to skip using a fork or a patch-on-install approach, you can select the parent commit (fcce6bce6a2f4b14eea7ea388031c0aa65e4b67d) instead, you're probably not missing out on much; only https://github.com/chriskempson/base16-vim/pull/187/commits/cbc9d08f80bb5c985fa8feadf8b6cff35f61b05d and https://github.com/chriskempson/base16-vim/pull/186/commits/83e8870e617b50c4e1284ccbbab82a517b6c97c1.

**Comment by benmezger** [comment]
I find it impressive how @chriskempson can maintain such repository without ever replying to such a huge thread

**Comment by sonnysideup** [comment]
So what's the outcome here then? I see a bunch of workarounds but perhaps I overlooked the fix?

**Comment by williamsmj** [comment]
It was fixed in #198. No workarounds necessary. Just upgrade to the current version of the package.

**Comment by chriskempson** [comment]
@benmezger Apologies! Should have seen this sooner.
